pkg/vendir/fetch/hg/hg.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package hg
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	ctlconf "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/config"
+	ctlfetch "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/fetch"
+	ctlver "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/versions/v1alpha1"
+)
+
+type Hg struct {
+	opts       ctlconf.DirectoryContentsHg
+	infoLog    io.Writer
+	refFetcher ctlfetch.RefFetcher
+}
+
+func NewHg(opts ctlconf.DirectoryContentsHg,
+	infoLog io.Writer, refFetcher ctlfetch.RefFetcher) *Hg {
+
+	return &Hg{opts, infoLog, refFetcher}
+}
+
+// nolint:golint
+type HgInfo struct {
```

Comment:
is word "SHA" and "Commit" proper terms within mercurial for those concepts?

Reply:
`SHA` makes sense within hg, but `commits` are named `changeSets`, so `CsetTitle` is surely better here

Reply:
if you dont mind, lets use "longer" version: ChangeSetTitle, or ChangesetTitle. a bit easier to google around for for mercurial newbies.

Reply:
@cppforlife I've rename it to `ChangeSetTitle`.